File: src/test/java/edu/kit/iti/formal/stvs/view/JavaFxTest.java
package edu.kit.iti.formal.stvs.view;

import javafx.application.Application;
import javafx.geometry.Orientation;
import javafx.scene.Node;
import javafx.scene.Scene;
import javafx.scene.control.SplitPane;
import javafx.stage.Stage;

import java.util.List;
import java.util.function.Supplier;

/**
 * Created by Lukas on 20.01.2017.
 */
public class JavaFxTest extends Application {

  private static Supplier<Scene> toBeViewed;

  public static void main(String[] args) {
    launch(args);
  }

  public static void setToBeViewed(Supplier<Scene> toBeViewed) {
    JavaFxTest.toBeViewed = toBeViewed;
  }

  public static void runView(Supplier<Scene> toBeViewed) {
    setToBeViewed(toBeViewed);
    Application.launch(JavaFxTest.class);
  }

  public static void runSplitView(Supplier<List<Node>> supplierOfViews) {
    runView(() -> {
      SplitPane pane = new SplitPane();
      pane.getItems().addAll(supplierOfViews.get());

      Scene scene = new Scene(pane, 800, 600);

      scene.getStylesheets().add(JavaFxTest.class.getResource("style.css").toExternalForm());

      return scene;
    });
  }

  @Override
  public void start(Stage primaryStage) throws Exception {
    primaryStage.setScene(toBeViewed.get());
    primaryStage.show();
  }
}

File: stvs/src/test/kotlin/edu/kit/iti/formal/stvs/view/JavaFxTest.kt
/* *****************************************************************
 * This file belongs to verifaps-lib (https://verifaps.github.io).
 * SPDX-License-Header: GPL-3.0-or-later
 * 
 * This program isType free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program isType distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a clone of the GNU General Public
 * License along with this program.  If not, see
 * <http://www.gnu.org/licenses/gpl-3.0.html>.
 * *****************************************************************/
package edu.kit.iti.formal.stvs.view

import javafx.application.Application
import javafx.scene.Node
import javafx.scene.Scene
import javafx.scene.control.SplitPane
import javafx.stage.Stage
import java.util.function.Supplier

/**
 * Created by Lukas on 20.01.2017.
 */
class JavaFxTest : Application() {
    @Throws(Exception::class)
    override fun start(primaryStage: Stage) {
        primaryStage.scene = toBeViewed!!.get()
        primaryStage.show()
    }

    companion object {
        private var toBeViewed: Supplier<Scene?>? = null

        @JvmStatic
        fun main(args: Array<String>) {
            launch(*args)
        }

        fun setToBeViewed(toBeViewed: Supplier<Scene?>?) {
            Companion.toBeViewed = toBeViewed
        }

        fun runView(toBeViewed: Supplier<Scene?>?) {
            setToBeViewed(toBeViewed)
            launch(JavaFxTest::class.java)
        }

        fun runSplitView(supplierOfViews: Supplier<List<Node?>>) {
            runView {
                val pane = SplitPane()
                pane.items.addAll(supplierOfViews.get())

                val scene = Scene(pane, 800.0, 600.0)

                scene.stylesheets.add(JavaFxTest::class.java.getResource("style.css")!!.toExternalForm())
                scene
            }
        }
    }
}